packages/matter-node.js/src/matter/MatterController.ts

```diff
@@ -36,32 +38,48 @@ const ADMIN_VENDOR_ID = new VendorId(752);
 const logger = Logger.get("MatterController");
 
 export class MatterController {
-    public static async create(scanner: Scanner, netInterfaceIpv4: NetInterface, netInterfaceIpv6: NetInterface) {
-        const certificateManager = new RootCertificateManager();
+    public static async create(scanner: Scanner, netInterfaceIpv4: NetInterface, netInterfaceIpv6: NetInterface, persistenceManager: PersistenceManager) {
+        const certificateManager = new RootCertificateManager(persistenceManager);
+
         const ipkValue = Crypto.getRandomData(16);
         const fabricBuilder = new FabricBuilder(FABRIC_INDEX)
             .setRootCert(certificateManager.getRootCert())
             .setRootNodeId(CONTROLLER_NODE_ID)
             .setIdentityProtectionKey(ipkValue)
             .setRootVendorId(ADMIN_VENDOR_ID);
         fabricBuilder.setOperationalCert(certificateManager.generateNoc(fabricBuilder.getPublicKey(), FABRIC_ID, CONTROLLER_NODE_ID));
-        const fabric = await fabricBuilder.build();
-        return new MatterController(scanner, netInterfaceIpv4, netInterfaceIpv6, certificateManager, fabric);
+
+        // Check if we have a fabric stored in the persistence, if yes initialize this one, else build a new one
+        const controllerPersistance = persistenceManager.createPersistence("MatterController");
+        if (controllerPersistance.has("fabric")) {
+            const storedFabric = Fabric.fromJson(controllerPersistance.get<FabricJsonObject>("fabric"));
```

Comment:
This API requires knowledge of the storage format and raw parsing of JSON.  The storage API should simply provide values from field name for you.

Ideally, this would look something like `fabric = storage.get("fabric")`.

Reply:
I agree, I decided for the "builder" approach because we already had the builder used in the process, so yes was "lowest effort".  Alternatively we need to add logic to storage on how to persist and also instanciate kind of any of the classes we have because t is not just data but the instace of a Fabric class. We coukld do that by defining a "Persistable" interface and classes implement that by providing toJson method and fromJson-static-builder-methods, the the storage can do that. WDYT?